Add docs for multiservice ArgoCD applications support (#23963)
DataDog/documentation: content/en/continuous_delivery/deployments/argocd.md

    dd_customtags: "region:us1-east, team:backend"
```

## Tag an Argo CD application deploying multiple services

> In case your Argo CD application deploys more than one service, Datadog can automatically infer the services deployed
> from an application sync based on the Kubernetes resources that were modified.
> For this to work, you need to [monitor your Kubernetes infrastructure using the Datadog agent][14] and your Kubernetes resources should have the following labels:

Reviewer: ```suggestion
If your Argo CD application deploys more than one service, Datadog can automatically infer the services deployed from an application sync. Datadog infers the services based on the Kubernetes resources that were modified.

To enable automatic service tagging, you need to [monitor your Kubernetes infrastructure using the Datadog Agent][14] and your Kubernetes resources should have the following labels:
```

> 1. `service` (required): specifies the Datadog service of this resource
> 2. `team` (optional): specifies the Datadog team of this resource

Reviewer: ```suggestion
- `service` (required): specifies the Datadog service of this resource
- `team` (optional): specifies the Datadog team of this resource
```

> Only the Kubernetes resources with the following kinds are eligible: `Deployment`, `ReplicaSet`, `StatefulSet`, `Service`, `DaemonSet`, `Pod`, `Job`, `CronJob`.

Reviewer: ```suggestion
Only the Kubernetes resources with the following kinds are eligible: `Deployment`, `ReplicaSet`, `StatefulSet`, `Service`, `DaemonSet`, `Pod`, `Job`, and `CronJob`.
```

Add the following annotations to your Argo CD application:

> 1. `dd_multiservice`: should be set to `true`. This annotation specifies that Datadog should automatically infer the services deployed in a sync based on the changed Kubernetes resources.
> 2. `dd_k8s_cluster`: should be set to the name of the Kubernetes cluster that the Argo CD application is deploying to. The name should be the same that is reported in the [Kubernetes product][15].

Reviewer: ```suggestion
- `dd_multiservice`: `true`. This annotation specifies whether Datadog automatically infers the services deployed in a sync based on the changed Kubernetes resources.
- `dd_k8s_cluster`: set to the name of the Kubernetes cluster that the Argo CD application deploys to. The name must match the name reported in the [Datadog Kubernetes product][15].
```

[11]: https://app.datadoghq.com/organization-settings/api-keys
[12]: https://argo-cd.readthedocs.io/en/stable/operator-manual/notifications/subscriptions/
[13]: /tracing/service_catalog

> [14]: https://docs.datadoghq.com/containers/kubernetes/

Reviewer 2: ```suggestion
[14]: /containers/kubernetes
```
